arch/RISCV/riscv_decode.gen.inc

```diff
@@ -0,0 +1,7056 @@
+#include <stdint.h>
+
+#include <stddef.h>
+
+#include <string.h>
+#include "riscv_ast.gen.inc"
+
+
+void decode(struct ast *tree, uint64_t binary_stream) {
+    uint64_t op = 0xFFFFFFFFFFFFFFFF ;
+    switch (binary_stream & 0x000000000000007F) {
+      case 0x37: 
+            op = RISCV_LUI ;
+      break; 
+      case 0x17: 
+            op = RISCV_AUIPC ;
+      break; 
+ 
+    }
+    if (op != 0xFFFFFFFFFFFFFFFF) { 
+        uint64_t rd = (binary_stream & 0x0000000000000F80)>>7 ;
```

Comment:
Please implement a helper function like `inline uint64_t get_bit_field64(uint64_t bytes, size_t start, size_t n);` for those. It can be in `utils.h`

Reply:
These helper functions are important because they allow bit extraction of `>64bit` instructions. The currently generated code has a maximum of 64bit supported.